src/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.sol

```diff
@@ -306,6 +304,14 @@ abstract contract BondingCurveBase_v1 is IBondingCurveBase_v1, Module_v1 {
         uint _issuanceTokenAmount
     ) internal virtual;
 
+    /// @notice Virtual function to handle collateral tokens before a buy.
+    /// @param  _provider The address from which the collateral tokens
```

Comment:
Why `_provider` instead of just `_sender`? "Provider" to me has a different connotation, but may just be me. Technically of course they "provide" the tokens, but sounds like a donation/provision mechanism instead of just a sending of funds?

Reply:
Thats the idea,
by abstracting it we actually enable the implementation of potential provision mechanisms instead of "just" sending funds :D